backend/src/main/java/com/example/dontwastetomuch/user/MyUserService.java

```diff
@@ -0,0 +1,41 @@
+package com.example.dontwastetomuch.user;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class MyUserService {
+
+
+    private final MyUserRepo userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public void createNewUser(MyUserCreationData userCreationData) {
+        if (!Objects.equals(userCreationData.getPassword(), userCreationData.getPasswordRepeat())) {
+            throw new IllegalArgumentException("passwords do not match");
+        }
+
+        if (userCreationData.getUsername() == null || userCreationData.getUsername().isBlank()) {
+            throw new IllegalArgumentException("username is blank");
+        }
+
+        MyUser user = new MyUser();
+        user.setUsername(userCreationData.getUsername());
+        user.setPassword(passwordEncoder.encode(userCreationData.getPassword()));
+
+        userRepository.save(user);
```

Comment:
Hier fehlen noch die Roles in zeile 30:
```Java
user.setRoles(List.of("user"))
```
Das geht natürlich nur wenn du in der MyUser Klasse vorher die Variable ``List<String> Roles;` definiert hast.